analyzer/habitat/analysis/operation.py

```diff
@@ -91,12 +92,18 @@ def __init__(
         measured_operation,
         forward,
         backward,
-        device
+        device,
+        measured_local=0,
```

Comment:
* `measured_local` belongs in `self.measured_operation`. 
* `predicted_local` can be derived directly from `self.measured_operation` by calling `to_device` on it - no need to store a redundant copy.
* Similarly, `unscaled_predicted` can be derived from existing class members, so it need not be stored.

Hence we can eliminate all three additional init parameters, and instead include the implementation in subsequent getter functions.